### DISCUSSION TITLE
aws-sdk-go-v2 traces and metrics

### DISCUSSION THREAD

**Discussion by rahanar** [aws/aws-sdk-go-v2#2806] [Q&A] [answered]
Hi there,

I see PR to add SDK metrics / spans was merged last week and it is a part of the latest release https://github.com/aws/aws-sdk-go-v2/pull/2798

Just would like to clarify a few things:
1. Based on the code merged, it looks like the changes were only implemented for S3 service. Is that correct?
2. If that's correct, when will other services be added (such as dynamoDB)?
3. Is there any documentation on how to enable these metrics?

Thank you

**Comment by lucix-aws** [comment, marked as answer]
They were added for every service. You were likely looking at one of the development PRs that only re-generated the s3 client.

In terms of enabling them -- they're technically on right now, but the trace and meter providers default to no-op implementations.

We separately released concrete OTEL SDK adapters you can hook up to clients, which is going to be the easiest way to actually capture traces/metrics and send them somewhere, I'm planning on getting an announcement post out with more detailed instructions on how to do this tomorrow.

* https://pkg.go.dev/github.com/aws/smithy-go/tracing/smithyoteltracing
* https://pkg.go.dev/github.com/aws/smithy-go/metrics/smithyotelmetrics

In advance of a formal announcement, the package docs for each of these have example code that shows how they're used.

**Comment by lucix-aws** [comment]
(consolidated my response, see above)

**Comment by rahanar** [comment]
awesome, thank you so much! looking forward to the announcement too

**Comment by lucix-aws** [comment]
https://github.com/aws/aws-sdk-go-v2/discussions/2810
